File: 백준/Silver/10845. 큐/큐.swift
import Foundation

class Queue {
    private var queue = [Int]()
    
    
    func push(x: Int) {
        queue.append(x)
    }
    
    func pop() {
        
        if queue.isEmpty {
            print(-1)
        } else {
            print(queue.removeFirst())
        }
    }
    
    func size() {
        print(queue.count)
    }
    
    func empty() {
        queue.isEmpty ? print(1) : print(0)
    }
    
    func front() {
        if queue.isEmpty {
            print(-1)
        } else {
            print(queue.first!)
        }
    }
    
    func back() {
        if queue.isEmpty {
            print(-1)
        } else {
            print(queue.last!)
        }
    }
}

let line = Int(readLine()!)!
let queue = Queue()

for _ in 0..<line {
    let order = readLine()?.split(separator: " ").map { $0 }
    
    let detailOrder = order?[0]
    
    switch detailOrder {
    case "push":
        if let num = Int(order?[1] ?? "") {
            queue.push(x: num)
        }
    case "pop":
        queue.pop()
    case "front":
        queue.front()
    case "back":
        queue.back()
    case "size":
        queue.size()
    case "empty":
        queue.empty()
    default:
        break
    }
}

File: 백준/Silver/10845. 큐/큐.cc
#include <iostream>
#include <queue>

using namespace std;

int main()
{
    ios_base::sync_with_stdio(0);
    cin.tie(0);
    
    queue<int> q;

    int n;
    cin >> n;
    
    while(n--) {
        string input;
        cin >> input;
        
        if(input == "push") {
            int add;
            cin >> add;
            q.push(add);
            
        } else if(input == "front") {
            if(q.empty())
                cout << -1 << "\n";
            else
                cout << q.front() << "\n";
        } else if(input == "back") {
            if(q.empty())
                cout << -1 << "\n";
            else
                cout << q.back() << "\n";
        } else if(input == "pop") {
            if(q.empty()) {
                cout << -1 << "\n";
            } else {
                cout << q.front() << "\n";
                q.pop();
            }
        } else if(input == "size") {
            cout << q.size() << "\n";
        } else if(input == "empty") {
            if(q.empty())
                cout << 1 << "\n";
            else
                cout << 0 << "\n";
        }
    }
    return 0;
}
